.github/workflows/python.yml

```diff
@@ -134,69 +201,54 @@ jobs:
-              with:
-                  os: ${{ matrix.host.OS }}
-                  target: ${{ matrix.host.TARGET }}
-                  github-token: ${{ secrets.GITHUB_TOKEN }}
-                  engine-version: ${{ matrix.engine.version }}
-        
-            - name: Test pubsub with pytest
-              working-directory: ./python
-              run: |
-                  source .env/bin/activate
-                  cd python/tests/
-                  pytest --asyncio-mode=auto -k test_pubsub --html=pytest_report.html --self-contained-html
-        
-            - name: Upload test reports
-              if: always()
-              continue-on-error: true
-              uses: actions/upload-artifact@v4
-              with:
-                  name: pubsub-test-report-python-${{ matrix.python }}-${{ matrix.engine.type }}-${{ matrix.engine.version }}-${{ matrix.host.RUNNER }}
-                  path: |
-                      python/python/tests/pytest_report.html
+    test-pubsub-python-client:
+      runs-on: ${{ matrix.host.RUNNER }}
+      needs: [load-engine-matrix, create-version-matrix, load-host-matrix]
+      timeout-minutes: 35
+      strategy:
+          fail-fast: false
+          matrix:
+              engine: ${{ fromJson(needs.load-engine-matrix.outputs.matrix) }}
+              python: ${{ fromJson(needs.create-version-matrix.outputs.matrix).python }}
+              host: ${{ fromJson(needs.load-host-matrix.outputs.matrix).include }}
+      steps:
+          - uses: actions/checkout@v4
+            with:
+                submodules: recursive
+      
+          - name: Set up Python
+            uses: actions/setup-python@v4
+            with:
+                python-version: ${{ matrix.python }}
+      
+          - name: Build Python wrapper
+            uses: ./.github/workflows/build-python-wrapper
+            with:
+                os: ${{ matrix.host.OS }}
+                target: ${{ matrix.host.TARGET }}
+                github-token: ${{ secrets.GITHUB_TOKEN }}
+                engine-version: ${{ matrix.engine.version }}
+      
+          - name: Test pubsub with pytest
+            working-directory: ./python
+            run: |
+                source .env/bin/activate
+                cd python/tests/
+                pytest --asyncio-mode=auto -k test_pubsub --html=pytest_report.html --self-contained-html
+      
+          - name: Upload test reports
+            if: always()
+            continue-on-error: true
+            uses: actions/upload-artifact@v4
+            with:
+                name: pubsub-test-report-python-${{ matrix.python }}-${{ matrix.engine.type }}-${{ matrix.engine.version }}-${{ matrix.host.RUNNER }}
+                path: |
+                    python/python/tests/pytest_report.html
 
     lint:
         runs-on: ubuntu-latest
+        needs: check_should_run
+        if: ${{ needs.check_should_run.outputs.should_run == 'true' }}
```

Comment:
May we need to run it always, since PR may contain changes in `lib.rs` of the python client

Reply:
Why its make any different?